workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ScreenOverlayDialogFactory.kt

```diff
@@ -159,8 +144,12 @@ public open class ScreenOverlayDialogFactory<S : Screen, O : ScreenOverlay<S>>(
     val contentViewHolder =
       buildContent(contentViewFactory, initialRendering.content, initialEnvironment, context)
 
-    return buildDialogWithContent(contentViewHolder).let { dialog ->
-      val window = requireNotNull(dialog.window) { "Dialog must be attached to a window." }
+    return buildDialogWithContent(
+      initialRendering,
+      initialEnvironment,
+      contentViewHolder
+    ).also { holder ->
```

Comment:
again nit: `apply` are we using other `this` references on `ScreenOverlayDialogFactory` that we need to keep `this` as that?

Reply:
I generally find `apply` and `with` blocks pretty confusing, after a few lines I quickly lose track of that `this` is.

Like in this case, there is only a single use of `holder`, everything else is side effect. I find that a lot more apparent with this usage.

Reply:
And even that one usage is strictly for getting the window, not for changing anything on the holder itself.

Reply:
OK ya that's valid.